### DISCUSSION TITLE
Add user selected directory find/search support?

### DISCUSSION THREAD

**Discussion by briandipalma** [LazyVim/LazyVim#681] [Ideas]
Sometimes I want to search for a file or grep for a string within a specific directory and neither the `cwd` or the `root` directory searches are specific enough. I was wondering if it would be OK to add https://github.com/princejoogie/dir-telescope.nvim as a dependency and add the `<leader>fd` mapping and move* `<leader>sd/D` mappings for directory specifying finds/searches?

* Controversial to change mappings I know but IMO searching diagnostics seems a rare case compared to grepping within a specific directory?

**Comment by dpetka2001** [comment]
Well searching diagnostics could also go to `x` which is used for diagnostics. When I first was searching for diagnostics through Telescope and was not yet used to Lazyvim keymaps, my first thought was to search them under `x` diagnostics and not `s` search. So I wouldn't be opposed to such a change should your proposal go through.

**Reply by briandipalma** [reply]
I was thinking `x` too given the `<leader>x` menu.

**Comment by Jomik** [comment]
You could easily make the change in your own config and see how it works, then open a PR.
I have not had the need to grep within a specific directory that was not `root` or `cwd` yet 🤔 
I suppose you are getting too many matches when searching from `cwd` and you want to narrow that down?

**Reply by briandipalma** [reply]
I have, I use it all the time. I work in a monorepo with half a dozen apps and a hundred or so packages so I find it invaluable.

**Reply by folke** [reply]
Yeah, this should go in your personal config. Not interested in having this installed for everyone

**Reply by briandipalma** [reply]
So it's the fact it needs a plugin that's the issue? Or is it the idea that you can specify a directory/select several different directories using `Tab` to search in that's the issue?

**Reply by Jomik** [reply]
@briandipalma Nothing prevents you from adding this to your own configuration.
No need to change LazyVim for this.
It is a very specific use-case - big monorepos. And it has other ways to work around it - changing your `cwd` or `root`, which is how I handle it, and feels nicer to me 😄

**Reply by briandipalma** [reply]
I also use it to search in multiple directories at the same time, I hit tab to select multiple directories and they all get used as part of the search. So I can grep in 3 or 4 directories called `components` by searching for `components` tabbing the ones I'm interested in hitting enter and then searching for the string. Changing my `cwd` over and over again is painful and I don't get an aggregated result view either.

**Reply by dpetka2001** [reply]
Well this isn't something to fret over. Everyone can have his own configuration and maintainer has the right to decide what Lazyvim will offer by default. Everyone happy this way since we can customize our configuration however we see fit.

**Reply by Jomik** [reply]
I guess the solution, as we have written multiple times, is to simply configure it in your own configuration. As we do with everything else that we need to change to suit our specific needs.

**Comment by talgat-ruby** [comment]
I am also missing it. Currently I use search and replace, without actual replace.

**Reply by briandipalma** [reply]
Since the move to fzf-lua I've been making do with the fzf syntax to narrow searches down. It's not that bad, have you tried searching for a string, hitting `ctrl+g` and then narrowing down the results that way?

**Reply by talgat-ruby** [reply]
Let me try, thanks.
